src/ir/auth_logic/is_name_constant.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/string_view.h"
+
+#ifndef SRC_IR_AUTH_LOGIC_IS_NAME_CONSTANT_H_
+#define SRC_IR_AUTH_LOGIC_IS_NAME_CONSTANT_H_
+
+// Determine if an identifier (which may be an argument of a predicate
+// or a principal name) is a constant or variable.
+// TODO(#673 aferr) it would be better to refactor the AST so that
+// identifiers are a separate AST node with constant and non-constant
+// children determined at parse time rather than here since
+// this information is available at parse time.
+bool IsNameConstant(absl::string_view id) {
```

Comment:
If this is only used in type_environment.cc, simply move it there? 

Otherwise, if you need keep this in the header file, you should add some tests.

Reply:
It's also used in `universe_relation_insertion.cc`, so I needed to put it somewhere that it's accessible for both. It's both an implementation detail I'd rather hide and something I would really rather get rid of eventually. Is there another way to make this available to both `type_environment.cc` and `universe_relation_insertion.cc` and not put it in a header? I could duplicate it in both cc files but that seems worse.

Reply:
added the tests